src/declarative/components/analog/JoystickSensor.tsx

```diff
@@ -0,0 +1,75 @@
+import { attachAnalog } from '../../../procedure/analog/uniqueDevice/analog'
+import type { AnalogPin } from '../../../procedure/types/analog/analog'
+import { board } from '../../../procedure/utils/board'
+import React, { createContext, useEffect, useState } from 'react'
+import type { SerialPort } from 'serialport'
+
+export const JoystickSensorContext = createContext<SerialPort | null>(null)
+
+type JoystickSensorProps = {
+  pinx: AnalogPin
+  piny: AnalogPin
```

Comment:
Please name words like pinx and piny correctly in CamelCase!
Example: pinx => pinX 🙏